### PR TITLE
feat(automate): New Automate SDK

### PR DESCRIPTION
**Pull request by gjedlicska**
Adds new projects for `Speckle.Automate.SDK` and its respective tests.

Tests have been integrated into the CI `test-core` job, and a preliminary nuget release was made and tested.

### CODE REVIEW

File: Core/Core/Api/GraphQL/Client.GraphqlCleintOperations/Client.BranchOperations.cs [hunk 1]
```diff
@@ -160,6 +160,57 @@ public Task<Branch> BranchGet(string streamId, string branchName, int commitsLim
...
   }
 
+  /// <summary>
+  /// Gets a given model from a project.
+  /// </summary>
+  /// <param name="cancellationToken"></param>
+  /// <param name="projectId">Id of the project to get the model from</param>
+  /// <param name="modelId">Id of the model</param>
+  /// <returns></returns>
+  public async Task<Branch> ModelGet(
```
Comment (JR-Morgan) [inline Core/Core/Api/GraphQL/Client.GraphqlCleintOperations/Client.BranchOperations.cs:170]: Cancellation token as last parameter
Dev has been updated with this change propagated to others

File: Core/SpeckleAutomate/Schema.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
+# nullable enable
+using System;
+using System.Collections.Generic;
+
+namespace SpeckleAutomate;
+///<summary>
+///Values of the project, model and automation that triggere this function run.
+///</summary>
+public struct AutomationRunData
```
Comment (JR-Morgan) [inline Core/SpeckleAutomate/Schema.cs:9]: Readonlystruct?

**Review by JR-Morgan — COMMENTED** [review]
Just some comments, none important to address for now, but a round of polish before we merge to dev is a good idea

**Comment by AlanRynne** [comment]
Changed target branch to `dev` instead of `main`

File: Core/Core/Api/GraphQL/Client.GraphqlCleintOperations/Client.BranchOperations.cs (hunk 1 above)
Comment (JR-Morgan) [reply Core/Core/Api/GraphQL/Client.GraphqlCleintOperations/Client.BranchOperations.cs:136]: Resolved in my pr #2984

File: Core/SpeckleAutomate/SpeckleAutomate.csproj [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net7.0</TargetFramework>
```
Comment (JR-Morgan) [inline Core/SpeckleAutomate/SpeckleAutomate.csproj:4]: net6 or netstandard2.0 may be preferred.
We should discuss before pushing any nuget.

Comment (AlanRynne) [reply Core/SpeckleAutomate/SpeckleAutomate.csproj:4]: Discussed with Jedd and my take is that we should prefer netstandard2.0 unless higher is required

Comment (AlanRynne) [reply Core/SpeckleAutomate/SpeckleAutomate.csproj]: Changed :)

**Review by AlanRynne — APPROVED** [review]
Pending an incoming PR to update the example repo, this PR is good to go

**Comment by AlanRynne** [comment]
Example repo has been updated (currently using a pre-release version). Will be finally updated again once the stable 2.17 is out.

Merging this in so it will be released with everything else

**Pull request merged**